### PR TITLE
[ERSSUP-55130]-[]-[OAS documentation for ERSSUP-54157: APIM User calls e-RS Endpoint to Book or Defer Appointment]-[HR]

### PR DESCRIPTION
**Pull request by nhsd-hubert-raich**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [x] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [x] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [x] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: sandbox/src/mocks/PasRespondedWithErrorOperationOutcome.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      "details": {
+        "coding": [
+          {
+            "system": "https://fhir.nhs.uk/STU3/CodeSystem/eRS-APIErrorCode-1",
+            "code": "PAS_RESPONDED_WITH_ERROR",
+            "display": "Error recieved from PAS"
+          }
+        ]
+      },
+      "diagnostics": "PAS unable to create booking into slot '11002026' for UBRN: 000000070000. PAS Request Status: ERROR"
```
Comment (nhsd-jack-wainwright) [inline sandbox/src/mocks/PasRespondedWithErrorOperationOutcome.json:21]: It's probably not great that we're leaking the PAS request status out from our APIs, might be worth creating a tech debt around removing this from the diagnostic message.

File: sandbox/src/mocks/bookOrDeferAppointment/requests/DeferralWithSlotReference.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  },
+  "language": "en",
+  "resourceType": "Appointment",
+  "status": "proposed",
+  "reason": [
+    {
+      "coding": [
+        {
+          "system": "https://fhir.nhs.uk/STU3/CodeSystem/eRS-ReviewReason-1",
+          "code": "SLOT_NOT_AVAILABLE"
```
Comment (nhsd-jack-wainwright) [inline sandbox/src/mocks/bookOrDeferAppointment/requests/DeferralWithSlotReference.json:17]: I think this should be using the "BOOKING_ATTEMPT_PROBLEM" reason when including a slot. If I remember correctly the "SLOT_NOT_AVAILABLE" reason is when attempting to book to a service that doesn't have any slots available.

File: sandbox/src/mocks/bookOrDeferAppointment/requests/DeferralBookingAttemptProblem.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  },
+  "language": "en",
+  "resourceType": "Appointment",
+  "status": "proposed",
+  "reason": [
+    {
+      "coding": [
+        {
+          "system": "https://fhir.nhs.uk/STU3/CodeSystem/eRS-ReviewReason-1",
+          "code": "BOOKING_ATTEMPT_PROBLEM"
```
Comment (nhsd-jack-wainwright) [inline sandbox/src/mocks/bookOrDeferAppointment/requests/DeferralBookingAttemptProblem.json:17]: I think when supplying the "BOOKING_ATTEMPT_PROBLEM" reason you also need to provide the reference to the slot you were attempting to book to.

Comment (nhsd-jack-wainwright) [reply sandbox/src/mocks/bookOrDeferAppointment/requests/DeferralBookingAttemptProblem.json:17]: Ah nope sorry I'm thinking of the SLOT_NOT_AVAILABLE case.

File: sandbox/src/mocks/bookOrDeferAppointment/requests/DeferralWithSlotReference.json (hunk 2 above)
Comment (nhsd-jack-wainwright) [reply sandbox/src/mocks/bookOrDeferAppointment/requests/DeferralWithSlotReference.json:17]: Looking at this again this looks to be correct as it is the "SLOT_NOT_AVAILALBLE" where the client is expected to provide the details of the slot they attempted to book to.

File: specification/components/schemas/eRS-Appointment.yaml [hunk 4]
```diff
@@ -46,18 +45,79 @@ properties:
       - cancelled
       - noshow
     example: proposed
+  reason:
```
Comment (nhsd-david-wass) [inline specification/components/schemas/eRS-Appointment.yaml:48]: we only accept 'booked' or 'proposed' here - I wonder whether we need an explanation

File: specification/components/schemas/responses/bookOrDeferAppointment/422Response.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+description: |
+  Where status code 422 (Unprocessable Entity) is returned then an eRS-OperationOutcome-1 will be included in the body, as detailed below. 
+  Check diagnostics property for specific information regarding the error.
+  
+  | Error code                         | Description                                                                                                                  |
+  | ---------------------------------- | ---------------------------------------------------------------------------------------------------------------------------- |
+  | INVALID_VALUE                      | The input provided does not conform to the expected data types and format. |
+  | REFERENCE_NOT_FOUND                | A supplied reference could not be resolved to valid resource (e.g. a patient, clinician or an organisation). |
+  | NO_RELATIONSHIP                    | The user does not have a legitimate relationship with the request.    |
```
Comment (nhsd-david-wass) [inline specification/components/schemas/responses/bookOrDeferAppointment/422Response.yaml:9]: For appointment status, we return INVALID_CODE if it is not one of the enumeration, but we also return INVALID_VALUE if it is not either 'booked' or 'proposed'

Comment (nhsd-hubert-raich) [reply specification/components/schemas/responses/bookOrDeferAppointment/422Response.yaml:10]: I've added INVALID_CODE  to the list now, good catch :)

File: specification/components/schemas/eRS-Appointment.yaml [hunk 6]
```diff
@@ -70,6 +132,7 @@ properties:
     maxItems: 1
   created:
     type: string
+    nullable: true
```
Comment (nhsd-jack-wainwright) [inline specification/components/schemas/eRS-Appointment.yaml:135]: I think an Appointment should always have a created date time, do we sometimes not include this field?

**Review by nhsd-jack-wainwright — COMMENTED** [review]
Looks good to me :), just a couple minor comments.

File: specification/components/schemas/eRS-Appointment.yaml (hunk 6 above)
Comment (nhsd-hubert-raich) [reply specification/components/schemas/eRS-Appointment.yaml:135]: Yes, it not included (or checked) on book flow as it gets generated on the server side. We actually send it as null from our regression pack.

**Pull request merged**